src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -689,6 +689,23 @@ public static K8SConfiguration LoadKubeConfig(Stream kubeconfigStream)
             return LoadKubeConfigAsync(kubeconfigStream).GetAwaiter().GetResult();
         }
 
+        /// <summary>
+        ///     Saves Kube Config
+        /// </summary>
+        /// <param name="kubeconfig">Instance of the <see cref="K8SConfiguration"/> class to save</param>
+        /// <param name="kubeconfigPath">Path to save the kubeconfig file or <see cref="KubeConfigDefaultLocation"/> if not provided</param>
+        public static void SaveKubeConfig(K8SConfiguration kubeconfig, string kubeconfigPath = null)
+        {
+            if (kubeconfig == null)
+            {
+                throw new NullReferenceException(nameof(kubeconfig));
+            }
+
+            kubeconfigPath ??= kubeconfig.FileName ?? KubeConfigDefaultLocation;
+
+            File.WriteAllTextAsync(kubeconfigPath, Yaml.SaveToString(kubeconfig)).GetAwaiter().GetResult();
```

Comment:
I would suggest adding both `Save` and `SaveAsync` and then just calling `SaveAsync` from `Save` with an await.